core/src/test/java/eu/fasten/core/data/RevisionCallGraphTest.java

```diff
@@ -1,209 +1,204 @@
-		assertEquals("4.3.2", constraints.get(0).upperBound);
-		constraints = depset.get(1).get(1).constraints;
-		assertEquals(1, constraints.size());
-		assertEquals("1.1", constraints.get(0).lowerBound);
-		assertEquals("2.0", constraints.get(0).upperBound);
+    }
+
+    @Test
+    public void testCallGraph() throws JSONException {
+        String callGraph = "{\n" +
+                "    \"forge\": \"mvn\",\n" +
+                "    \"product\": \"foo\",\n" +
+                "    \"version\": \"2.0\",\n" +
+                "    \"generator\": \"OPAL\",\n" +
+                "    \"depset\":\n" +
+                "      [\n" +
+                "        [{ \"forge\": \"mvn\", \"product\": \"a\", \"constraints\": [\"[1.0..2.0]\", \"[4.2..]\"]}],\n" +
+                "        [{ \"forge\": \"other\", \"product\": \"b\", \"constraints\": [\"[4.3.2]\"]},\n" +
+                "        { \"forge\": \"other\", \"product\": \"c\", \"constraints\": [\"[1.1..2.0]\"]}]\n" +
+                "      ],\n" +
+                " \"cha\": {\n" +
+                "     \"/my.package/A\": {\n" +
+                "       \"methods\": {\n" +
+                "         \"1\": \"/my.package/A.f(A)B\",\n" +
+                "         \"2\": \"/my.package/A.g(A,%2F%2Fjdk%2Fjava.lang%2Fint)%2F%2Fjdk%2Fjava.lang%2Fint\",\n" +
+                "       },\n" +
+                "       \"superInterfaces\": [],\n" +
+                "       \"sourceFile\": \"A.java\",\n" +
+                "       \"superClasses\": [\n" +
+                "         \"/java.lang/Object\"\n" +
+                "       ]\n" +
+                "     },\n" +
+                "   \"/my.other.package/C\": {\n" +
+                "     \"methods\": {\n" +
+                "       \"3\": \"/my.other.package/C.g(%2Fmy.package%2FA)%2Fmy.package%2FB\",\n" +
+                "     },\n" +
+                "     \"superInterfaces\": [],\n" +
+                "     \"sourceFile\": \"C.java\",\n" +
+                "     \"superClasses\": [\n" +
+                "       \"/java.lang/Object\"\n" +
+                "     ]\n" +
+                "   }\n" +
+                "},\n" +
+                " \"graph\": {\n" +
+                "     \"internalCalls\": [\n" +
+                "       [\n" +
+                "         1,\n" +
+                "         3\n" +
+                "       ]\n" +
+                "     ],\n" +
+                "     \"externalCalls\": [\n" +
+                "       [\n" +
+                "         \"2\",\n" +
+                "         \"//b/their.package/TheirClass.method(TheirOtherClass)TheirOtherClass\",\n" +
+                "         {\n" +
+                "           \"invokevirtual\":\"1\"\n" +
+                "         }\n" +
+                "       ]\n" +
+                "     ]\n" +
+                "   }\n" +
+                "}";
+        JSONObject json = new JSONObject(callGraph);
+        RevisionCallGraph cg = new RevisionCallGraph(json);
```

Comment:
This is an interesting construct... why is the `JSONObject` required to instantiate the `RevisionCallGraph`? Wouldn't it be better to leave the parsing logic outside of the data structure and better use, for example, `GSON` or `Jackson` for the deserialization of the JSON string?

Reply:
Now when I think about it, refactoring the RCG and it's tests seems like a nice idea for the next PR, especially because (as far as I'm know) our partners are planning to create language-specific RCGs, so we need a good base class.